pkg/kv/dist_sender.go

```diff
@@ -59,6 +60,39 @@ const (
 	defaultSenderConcurrency = 500
 )
 
+var (
+	// metaDistSenderTotals identify the total rpc sends of DistSender.
+	metaDistSenderTotals = metric.Metadata{Name: "distsender.totals"}
+	// metaDistSenderLocals identify the total local rpc sends of DistSender.
+	metaDistSenderLocals = metric.Metadata{Name: "distsender.locals"}
+	// metaDistSenderRetries identify the total retry times DistSender.
+	metaDistSenderRetries = metric.Metadata{Name: "distsender.retries"}
+	// metaDistSenderNotLeaseHolderErrs identify total NotLeaseHolderErrs of DistSender.
+	metaDistSenderNotLeaseHolderErrs = metric.Metadata{Name: "distsender.notLeaseHolderErrs"}
+)
+
+// DistSenderMetrics used to metric some statics for debug.
+type DistSenderMetrics struct {
+	registry           *metric.Registry
+	total              *metric.Counter
```

Comment:
the individual metric fields have to be exported, or the registry won't be able to find them through reflection.